cd/ack-chart/update-chart.sh

```diff
@@ -0,0 +1,278 @@
+
+_get_semver_diff() {
+    local __current_version=$1
+    local __new_version=$2
+
+    local trimmed_current
+    local trimmed_new
+
+    # This assumes the version takes the form of `vX.Y.Z`
+    trimmed_current="$(echo "$__current_version" | cut -c2-)"
+    trimmed_new="$(echo "$__new_version" | cut -c2-)"
+
+    if [[ "$(echo "$trimmed_current" | cut -d"." -f1)" != "$(echo "$trimmed_new" | cut -d"." -f1)" ]]; then
+        echo "$DEPENDENCY_DIFF_MAJOR"
+    elif [[ "$(echo "$trimmed_current" | cut -d"." -f2)" != "$(echo "$trimmed_new" | cut -d"." -f2)" ]]; then
+        echo "$DEPENDENCY_DIFF_MINOR"
+    elif [[ "$(echo "$trimmed_current" | cut -d"." -f3)" != "$(echo "$trimmed_new" | cut -d"." -f3)" ]]; then
+        echo "$DEPENDENCY_DIFF_PATCH"
+    else
+        echo ""
+    fi
+}
+
+_increment_chart_version() {
+    local __current_version=$1
+    local __patch_diff=$2
+
+    local current_major
+    local current_minor
+    local current_patch
+
+    current_major="$(echo "$__current_version" | cut -d"." -f1)"
+    current_minor="$(echo "$__current_version" | cut -d"." -f2)"
+    current_patch="$(echo "$__current_version" | cut -d"." -f3)"
+
+    # Increment the largest diff number and reset any smaller parts
+    if [[ "$__patch_diff" == "$DEPENDENCY_DIFF_PATCH" ]]; then
+        current_patch="$((current_patch + 1))"
+    elif [[ "$__patch_diff" == "$DEPENDENCY_DIFF_MINOR" ]]; then
+        current_patch="0"
+        current_minor="$((current_minor + 1))"
+    elif [[ "$__patch_diff" == "$DEPENDENCY_DIFF_MAJOR" ]]; then
+        current_patch="0"
+        current_minor="0"
+        current_major="$((current_major + 1))"
+    fi
+
+    echo "${current_major}.${current_minor}.${current_patch}"
+}
+
+_add_chart_dependency() {
+    local __dependency_name=$1
+    local __dependency_version=$2
+    local __service_name=$3
+
+    NAME=$__dependency_name SERVICE_NAME=$__service_name VERSION=$__dependency_version \
+    yq --inplace '.dependencies += {
+        "name": env(NAME),
+        "alias": env(SERVICE_NAME),
+        "version": env(VERSION),
+        "repository": "oci://public.ecr.aws/aws-controllers-k8s",
+        "condition": (env(SERVICE_NAME) + ".enabled")
+    }' "$PARENT_CHART_CONFIG"
+}
+
+_upgrade_chart_dependency() {
+    local __dependency_name=$1
+    local __dependency_version=$2
+    local __service_name=$3
+
+    NAME=$__dependency_name yq --inplace 'del(.dependencies[] | select(.name == env(NAME)))' "$PARENT_CHART_CONFIG"
+
+    _add_chart_dependency "$__dependency_name" "$__dependency_version" "$__service_name"
+}
+
+_rebuild_chart_dependencies() {
+    local ecr_pw
+    ecr_pw=$(aws ecr-public get-login-password --region us-east-1)
+    echo "$ecr_pw" | helm registry login -u AWS --password-stdin public.ecr.aws
+
+    helm dependency update
+}
+
+_add_chart_values_section() {
+    local __service_name=$1
+
+    SERVICE_NAME=$__service_name yq --inplace '.[env(SERVICE_NAME)+"-chart"] += {
+        "enabled": false
```

Comment:
any reason not to enable the chart by default?

Reply:
Every chart needs IRSA and other configurations enabled, anyway. I figure if you're going to the trouble of adding those, you can enable it in the same values set